cmd/osm-controller/osm-controller.go

```diff
@@ -333,3 +337,44 @@ func getOSMControllerPod(kubeClient kubernetes.Interface) (*corev1.Pod, error) {
 
 	return pod, nil
 }
+
+// StartGlobalLogLevelHandler registers a listener to meshconfig events and log level changes,
+// and applies new log level at global scope
+func StartGlobalLogLevelHandler(cfg configurator.Configurator, stop <-chan struct{}) {
+	meshConfigChannel := events.GetPubSubInstance().Subscribe(
+		announcements.MeshConfigAdded,
+		announcements.MeshConfigDeleted,
+		announcements.MeshConfigUpdated)
+
+	// Run config listener
+	// Bootstrap after subscribing
+	currentLogLevel := constants.DefaultOSMLogLevel
+	logLevel := cfg.GetOSMLogLevel()
+	log.Info().Msgf("Setting initial log level from meshconfig: %s", logLevel)
+	err := logger.SetLogLevel(logLevel)
+	if err != nil {
+		log.Error().Msgf("Failed to set initial log level from meshconfig: %v", err)
+	} else {
+		currentLogLevel = logLevel
+	}
+
+	go func() {
+		for {
+			select {
+			case <-meshConfigChannel:
+				logLevel := cfg.GetOSMLogLevel()
+				if logLevel != currentLogLevel {
+					err := logger.SetLogLevel(logLevel)
+					if err != nil {
+						log.Error().Msgf("Failed to set log level from meshconfig: %v", err)
```

Comment:
I tend to use `Error` in log messages consistently instead of `Failed` so we can consistently grep for error messages in logs.
```suggestion
						log.Error().Msgf("Error updating log level to %s: %s", logLevel, err)
```

Reply:
good advice, changing

Reply:
will keep in mind also